sympy/sets/undefinedset.py

```diff
@@ -0,0 +1,122 @@
+from sympy.core import Expr
+from sympy.core.symbol import Str
+from sympy.core.sympify import _sympify
+from .sets import Set, FiniteSet
+
+__all__ = [
+    'UndefinedSet', 'SetElement'
+]
+
+class UndefinedSet(Set):
+    """
+    An abstract set with no condition defined, which is commonly
+    encountered in set theory and abstract algebra.
+
+    Parameters
+    ==========
+
+    name : str
+        Name of the undefined set.
+
+    supersets : set of Sets, optional
```

Comment:
I have thought that this also suffers from similar problem as the previous `CoordSystem` relations, that objects need to be defined first in order to define full relation.
Would there be any cases where some mutual relation can occur?

Reply:
In `CoordSystem`, mutual reference was crucial since SymPy cannot deduce the inverse transformation between curvilinear coordinates. But this superset-subset relation is simple, so mutual relation won't be needed, I suppose.

Reply:
Actually, I start to find that mathematicians often denote equality between sets as A is subset of B and B is subset of A.
I think that it’s be better to allow full relation diagram here, for this reason

Reply:
Sounds reasonable. Instead of defining relational diagram, how about introducing assumption predicate something like ``Q.subset(A, B)``?

Reply:
That can also be a valid idea

Reply:
My group is planning to implement a finitary relation class, involving equality and inequality, as a subclass of ``ProductSet``. After that is done and API with assumption module is established,